src/cpp/server/server_cc.cc

```diff
@@ -67,6 +67,15 @@ namespace {
 // max-threads set) to the server builder.
 #define DEFAULT_MAX_SYNC_SERVER_THREADS INT_MAX
 
+// Give a useful status error message if the resource is exhausted specifically
+// because the server threadpool is full.
+const char* kServerThreadpoolExhausted = "Server Threadpool Exhausted";
+
+// Although we might like to give a useful status error message on unimplemented
+// RPCs, it's not always possible since that also would need to be added across
+// languages and isn't actually required by the spec.
```

Comment:
Why doesn't the server threadpool exhausted error message need to be added across languages?

Reply:
Every language has its own API and implementation. They might each run into this condition in different ways. This is in contrast to the unimplemented method, which is language independent in the sense that it only depends on the service specification and the set of methods actually implemented. For what it's worth, I am okay with this PR being rejected. It was a user feature request to help with debugging, not a fundamental bug fix.

Reply:
I see the value of the PR since I can understand that being pretty hard to debug - I'm more wondering if there are other similar error messages we could add at this point, maybe not to this PR but generally? For instance, a user asked to specify more clearly whether proto parsing failure occurred - is there a way to detect this internally?

Reply:
Great idea for a sequence of follow-ons. I've created a new Github project called debuggability. We can also use create some tracking issues for that project based on the topics that @sheenaqotj left for us (as well as any others that come up)

WRT proto parsing: we detect that in our calls of `Deserialize` but probably don't give useful feedback. It is also controversial that we currently return that as status `INTERNAL` but some users (including some PTBs) would prefer that to be `INVALID_ARGUMENT`. That has a long history but I think we have approval for starting a migration of that to `INVALID_ARGUMENT`. That said, that would be a multi-quarter multi-language project.

Side note: this PR passes TGP so I don't think it hurts anybody that we know of.

Reply:
And thanks for the review!